### PR TITLE
932: Products

### PR DESCRIPTION
**Pull request by rimi-itk**
#### Link to ticket

https://leantime.itkdev.dk/#/tickets/showTicket/932

#### Description

Adds products.

#### Screenshot of the result

If your change affects the user interface you should include a screenshot of the result with the pull request.

#### Checklist

- [ ] My code is covered by test cases.
- [ ] My code passes our test (all our tests).
- [ ] My code passes our static analysis suite.
- [ ] My code passes our continuous integration process.

### CODE REVIEW

File: src/Form/IssueType.php [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+class IssueType extends AbstractType
+{
+    public function buildForm(FormBuilderInterface $builder, array $options): void
+    {
+        $builder
+            ->add('name')
```
Comment (tuj) [inline src/Form/IssueType.php:19]: I don't think any of these fields should be editable, so this file seems superfluous.

File: templates/client/index.html.twig [hunk 2]
```diff
@@ -21,7 +23,7 @@
     {{ form_end(form) }}
 
     <table class="table margin-bottom">
-        <thead class="table-thead">
```
Comment (tuj) [inline templates/client/index.html.twig:24]: why this change? does it not conflict with th (.table-th) element styling?

File: templates/invoices/edit.html.twig [hunk 3]
```diff
@@ -3,7 +3,11 @@
 {% block title %}{{ 'invoices.edit'|trans }}{% endblock %}
 
 {% block content %}
-    <h1 class="page-title">{{ invoice.name }}</h1>
+
```
Comment (tuj) [inline templates/invoices/edit.html.twig:6]: Remove empty lines

File: templates/issue/index.html.twig [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+    {% set number_of_columns = 5 %}
+
+    <table class="table margin-bottom">
+        <thead class="table-th">
+        <tr>
+            {{ macros.knp_pagination_sortable_header(issues, 'issue.id'|trans, 'issue.id') }}
+            {{ macros.knp_pagination_sortable_header(issues, 'issue.name'|trans, 'issue.name') }}
+            {{ macros.knp_pagination_sortable_header(issues, 'issue.status'|trans, 'issue.status') }}
+
+            {#
```
Comment (tuj) [inline templates/issue/index.html.twig:28]: remove

File: templates/product/index.html.twig [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+        </div>
+    </div>
+    {{ form_end(form) }}
+
+    {% set number_of_columns = 7 %}
+
+    <table class="table margin-bottom">
+        <thead class="table-th">
+        <tr>
+            <th class="table-th{% if products.isSorted('product.id') %} sorted{% endif %}">{{ knp_pagination_sortable(products, 'product.id'|trans, 'product.id') }}</th>
```
Comment (tuj) [inline templates/product/index.html.twig:25]: Shouldn't these (and all the other sortable th) use the macro knp_pagination_sortable_header ?

File: templates/issue/show.html.twig [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+                <tr class="table-tr">
+                    <td class="table-td" colspan="{{ number_of_columns }}">{{ 'issue.product.list_no_records_found'|trans }}</td>
+                </tr>
+            {% endif %}
+        </tbody>
+    </table>
+
+    {% if add_product_form %}
+        {# Important: The actual add_product_form *must* come after the individual form fields in the table #}
+        {# Furthermore, we must add rou custom id since the form contains a div element with the (default) form id #}
```
Comment (tuj) [inline templates/issue/show.html.twig:115]: rou => row ?

File: config/packages/security.yaml [hunk 7]
```diff
@@ -39,7 +39,7 @@ security:
         - { path: ^/admin/management-report/, roles: ROLE_REPORT }
         - { path: ^/admin/team-report/, roles: ROLE_REPORT }
         - { path: ^/admin/sprint-report/, roles: ROLE_REPORT }
-        - { path: ^/admin/project/, roles: ROLE_ADMIN }
+        - { path: ^/admin/project/, roles: [ROLE_ADMIN, ROLE_PRODUCT_MANAGER] }
```
Comment (tuj) [inline config/packages/security.yaml:42]: add
 - { path: ^/admin/products/, roles: ROLE_PRODUCT_MANAGER }

**Review by tuj — CHANGES_REQUESTED** [review]

File: config/packages/framework.yaml [hunk 8]
```diff
@@ -2,7 +2,7 @@
 framework:
     secret: '%env(APP_SECRET)%'
     #csrf_protection: true
-    http_method_override: false
+    http_method_override: true
```
[the remark this answers is not in the document]
Comment (rimi-itk) [reply config/packages/framework.yaml:5]: I was fooling around with `PUT` and `PATCH` requests at some point (cf. https://symfony.com/doc/current/routing.html#matching-http-methods), but I don't do that anymore. I'll clean up.

File: src/Form/IssueType.php (hunk 1 above)
Comment (rimi-itk) [reply src/Form/IssueType.php:19]: You're right. It has been autogenerated and will be removed manually.

File: config/packages/framework.yaml (hunk 8 above)
[the remark this answers is not in the document]
Comment (rimi-itk) [reply config/packages/framework.yaml:5]: To use `PUT` requests in IssueController.

File: README.md [hunk 9]
```diff
@@ -83,47 +87,65 @@ JIRA_API_SERVICE_DEFAULT_BOARD=<VALUE>
 Sync projects and accounts.
```
[the remark this answers is not in the document]
Comment (rimi-itk) [reply README.md:88]: We'll do that in another pull request.

**Review by tuj — APPROVED** [review]
